components/modelregistry/modelregistry.go

```diff
@@ -124,3 +137,28 @@ func (m *ModelRegistry) ReconcileComponent(ctx context.Context, cli client.Clien
 	}
 	return nil
 }
+
+//go:embed resources/servicemesh-member.tmpl.yaml
+var smmTemplate string
+
+func createServicemeshMember(ctx context.Context, cli client.Client, dscispec *dsciv1.DSCInitializationSpec, namespace *corev1.Namespace) error {
+	tmpl, err := template.New("servicemeshmember").Parse(smmTemplate)
+	if err != nil {
+		return fmt.Errorf("error parsing servicemeshmember template: %w", err)
+	}
+	builder := strings.Builder{}
+	err = tmpl.Execute(&builder, struct {
+		Namespace    string
+		ControlPlane *infrav1.ControlPlaneSpec
+	}{Namespace: namespace.Name, ControlPlane: &dscispec.ServiceMesh.ControlPlane})
```

Comment:
`dscispec.ServiceMesh` can be `nil`, so it's worth guarding for it at some place in MR reconcile.

Reply:
I have that check in the other PR #1165 , so will merge the two when they are approved.